src/mongo/editors/MongoFindOneResultEditor.ts

```diff
@@ -34,7 +34,7 @@ export class MongoFindOneResultEditor implements ICosmosEditor<IMongoDocument> {
         const cachedCollectionNode = cachedCollectionNodes.find((node) => node.treeItem.label === this._collectionName);
```

Comment:
You should be able to get rid of most of this code and just do `tree.findNode(this.id)`